api/pkg/auth/auth.go

```diff
@@ -0,0 +1,151 @@
+package auth
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/markbates/goth/gothic"
+	"github.com/tektoncd/hub/api/gen/log"
+	"github.com/tektoncd/hub/api/pkg/app"
+	"github.com/tektoncd/hub/api/pkg/db/model"
+	"gorm.io/gorm"
+)
+
+type service struct {
+	app.Service
+	api app.Config
+}
+
+type request struct {
+	db            *gorm.DB
+	log           *log.Logger
+	defaultScopes []string
+	jwtConfig     *app.JWTConfig
+}
+
+type ProviderList struct {
+	Provider      string
+	EnterpriseUrl string
+	ClientId      string
```

Comment:
I don't think that we need `EnterpriseUrl` and `ClientId` in `/providers` endpoint as UI needs to be aware of only the providers configured with oauth server. WDYT?

Reply:
Yes makes sense, so on the UI side I think we need to remove client id from env